src/snowflake/snowpark/_internal/data_source/utils.py

```diff
@@ -359,6 +362,7 @@ def process_completed_futures(thread_futures):
                         )
                 thread_futures.clear()  # Clear the set since all are cancelled
                 raise
+    return upload_to_sf_end_time
```

Comment:
do we need `upload_to_sf_end_time`? I feel we can just `return time.perf_counter()`